kbmd/door.c

```diff
@@ -425,6 +382,12 @@ kbmd_door_server(void *cookie, char *argp, size_t arg_size, door_desc_t *dp,
 	nvlist_t *req = NULL;
 	tdata_t *td = tdata_get();
 
+	/*
+	 * If any file descriptors were passed to us, we want to close
+	 * them or else they will hang out in kbmd until it exits.
+	 */
+	close_door_desc(dp, n_desc);
```

Comment:
If a fifo is passed in, it will stay open in kbmd even after the calling process exits. I ran into this when I was still trying to track the tty -- when kbmadm was being called from a script, while the code would correctly detect the fd wasn't a tty, and without closing it, the fifo would stay open in kbmd which would leave the shell script that was calling kbmadm hung. While the tty stuff is now gone, it seems possible something could still at least try to do something like that -- though it'd just be a DOS attack at worst (if the fds aren't closed).

An alternative might be to set `DOOR_RELEASE` on any passed in descriptors and let door_return do it, though that'd require a bit more rework to pass `dp` everywhere needed.